### PR TITLE
Tutorial: More testable code

### PR DESCRIPTION
**Pull request by killercup**

### CODE REVIEW

File: src/tutorial/testing.md [hunk 1]
```diff
@@ -82,20 +82,169 @@ Worse, it often deals with user input,
 reads files,
 and writes output.
 
+## Making you code testable
```
Comment (Aaronepower) [inline src/tutorial/testing.md]: Making **your** code testable

File: src/tutorial/testing.md [hunk 2]
```diff
@@ -82,20 +82,169 @@ Worse, it often deals with user input,
...
+
+There are two complementary approaches to testing functionality:
+Testing the small units that you build your complete application from,
+and testing the final application "from the outside".
+Let's begin with the first one.
```
Comment (Aaronepower) [inline src/tutorial/testing.md]: The two types of testing should probably be named here like 'Testing the small units that you build your complete application from, these are called "unit tests". There is also testing the final application "from the outside" called "black box tests".'

File: src/tutorial/testing.md [hunk 3]
```diff
@@ -82,20 +82,169 @@ Worse, it often deals with user input,
...
+
+To figure our what we should test,
```
Comment (Aaronepower) [inline src/tutorial/testing.md]: To figure **out** what we should test

**Comment by killercup** [comment]
Thanks for the feedback! Merging this now -- you can always leave addition feedback in #58!

**Pull request merged**